adoc-mode.el

```diff
@@ -3002,6 +2993,38 @@ or \\[adoc-toggle-inline-images]."
           (adoc-create-image-overlay file start end)
           )))))
 
+(defun adoc-image-overlays (&optional begin end)
+  "Return list of image overlays in region from BEGIN to END.
+BEGIN and END default to the buffer boundaries
+ignoring any restrictions."
+  (save-restriction
+    (widen)
+    (unless begin (setq begin (point-min)))
```

Comment:
Just use `let` binding here. The usage of `setq` for introducing local bindings is mostly an antipatern in Elisp.

Reply:
No, there is no local binding introduced through the `setq`. The local binding is already introduced through the function argument `begin`. The `setq` just modifies its value.

What you get through the additional `let`-form is an superfluous local variable `begin` hiding the binding of the other local `begin`.

This also is reflected by one more statement in the byte-code of the version with `let`:

Version with `setq`:

    (with-temp-buffer
      (disassemble (byte-compile
    		;;;; The function:
    		(lambda (&optional my-unin)
    		  (unless my-unin (setq my-unin 1))
    		  (message "my-unin: %s" my-unin) ;; Do just something.
    		  )
    		)
    	       ;;;;;;;;;;
    	       (current-buffer))
      (buffer-string))


Generated Byte-Code of the version with `setq`:

    byte code:
      args: (&optional my-unin)
    0	varref	  my-unin
    1	goto-if-not-nil 1
    4	constant  1
    5	varset	  my-unin
    6:1	constant  message
    7	constant  "my-unin: %s"
    8	varref	  my-unin
    9	call	  2
    10	return	  

Version with `let`:

    (with-temp-buffer
      (disassemble (byte-compile
    		;;;; The function:
    		(lambda (&optional my-unin)
    		  (let ((my-unin (or my-unin 1)))
    		    (message "my-unin: %s" my-unin) ;; Do just something.
    		    ))
    		)
    	       ;;;;;;;;;;
    	       (current-buffer))
      (buffer-string))

Generated Byte-Code of the version with `let`:

    byte code:
      args: (&optional my-unin)
    0	varref	  my-unin
    1	goto-if-not-nil-else-pop 1
    4	constant  1
    5:1	varbind	  my-unin
    6	constant  message
    7	constant  "my-unin: %s"
    8	varref	  my-unin
    9	call	  2
    10	unbind	  1
    11	return

Reply:
My bad - too much multitasking this morning. :D I'd still write something like `(setq begin (or begin (point-min)))`, as `unless` feels like an overkill for this usecase, but I'm fine either way.

Reply:
`(setq begin (or begin (point-min)))` has the superfluous assignment `(setq begin begin)`.
`(unless begin (setq begin (point-min)))` does not have that assignment.

Reply:
Reads better to me regardless, but it's not something I feel strongly about. Feel free to keep it as is.

Reply:
Both variants have almost the same frequency in `/usr/share/emacs/29.1/lisp`. Since I am lazy, I do not change it. Furthermore, the statement `(unless begin (setq begin ...))` is more straight-forward than `(setq begin (or begin ...))`. The first version says explicitly what is done while the second version is more like a trick.